modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -405,3 +421,58 @@ ModularGapConductanceConstraint::computeQpResidual(Moose::MortarType
   mooseError("We should never get here. We should have errored in the constructor.");
 #endif
 }
+
+void
+ModularGapConductanceConstraint::deduceGeometryParameters()
+{
+  Point position;
+  Real area = 0.0;
+  const auto my_pid = processor_id();
+
+  // build side element list as (element, side, id) tuples
+  const auto bnd = _mesh.buildActiveSideList();
+
+  std::unique_ptr<const Elem> side_ptr;
+  for (auto [elem_id, side, id] : bnd)
+    if (id == _primary_id)
+    {
+      const auto * elem = _mesh.elemPtr(elem_id);
+      if (elem->processor_id() != my_pid)
+        continue;
+
+      // update side_ptr
+      elem->side_ptr(side_ptr, side);
+
+      // area of the (linearized) side
+      const auto side_area = side_ptr->volume();
+
+      // position of the side
+      const auto side_position = side_ptr->true_centroid();
+
+      // sum up
+      position += side_position * side_area;
+      area += side_area;
```

Comment:
Does this work for, e.g., 1D Cartesian? I am thinking of the meaning of volume and area for a 1D element.  It doesn't have to work here since it's mortar, but just wondering about what to do with a node-to-segment formulation.

Reply:
`side_area` will be 1 if the element from which we are building the side is an EDGE element

Reply:
in the odd 1D mesh that looks like this

`bbbbbB----AaaaaaaaaaaaaaaA----Bbbbbb`

Where `-` is the gap, `A` and `B` are the primary and secondary sides, and `a` and `b` are the volumes.

Here the automatic deduction would give the correct location (in the middle of the `a` volume).